lib/data.dart

```diff
@@ -93,3 +101,25 @@ Future<TA> getTaById(String taId) {
     return new TA(taData['name'], taDoc.id);
   });
 }
+
+Future<Course> getCourseById(String courseId) {
+  return db.collection('courses').doc(courseId).get().then((courseDoc) {
+    var courseData = courseDoc.data();
+    var courseTas = courseTaIds(courseData);
+    return new Course(
+        courseData['name'], courseDoc.id, courseData['yearId'], courseTas);
+  });
```

Comment:
The question about `catch` block. Is it possible to pass here the id of the course that is not presented in the DB? If no then everything is ok, if it is possible then catch block returning null or smthg like this is required

Reply:
Btw the only question that really needs an answer, others are more like comments

Reply:
It is possible that the course document does not exist and it could lead to errors. This code relies on the fact that the courses displayed to the user already exist.

Reply:
fine